syntax = "proto3";

package kv;

service KVServer {
  rpc Put(KVRequest) returns (KVResponse) {}
  rpc GetLocal(KVRequest) returns (KVResponse) {}
  rpc GetGlobal(KVRequest) returns (KVResponse) {}
  rpc Delete(KVRequest) returns (KVResponse) {}
}

message KVRequest {
  string key     = 1;
  bytes  value   = 2;
  uint64 version = 3;
  uint32 operator = 4;
}

message KVResponse {
  KVData data = 1;
  int32  ret  = 2;
}

message KVData {
  bytes  value      = 1;
  uint64 version    = 2;
  bool   is_deleted = 3;
}
